collector/ipvs.go

```diff
@@ -0,0 +1,258 @@
+			backend.remoteAddress.String(),
+			strconv.FormatUint(uint64(backend.remotePort), 10),
+			backend.proto,
+		}
+		c.backendConnectionsActive.WithLabelValues(labelValues...).Set(float64(backend.activeConn))
+		c.backendConnectionsInAct.WithLabelValues(labelValues...).Set(float64(backend.inActConn))
+		c.backendWeight.WithLabelValues(labelValues...).Set(float64(backend.weight))
+	}
+
+	c.backendConnectionsActive.Collect(ch)
+	c.backendConnectionsInAct.Collect(ch)
+	c.backendWeight.Collect(ch)
+
+	return nil
+}
+
+func getIpvsBackendStatus() ([]ipvsBackendStatus, error) {
+	file, err := os.Open(procNetIpvs)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	return parseIpvsBackendStatus(file)
+}
+
+func parseIpvsBackendStatus(file io.Reader) ([]ipvsBackendStatus, error) {
+	var (
+		status       = []ipvsBackendStatus{}
+		scanner      = bufio.NewScanner(file)
+		proto        string
+		localAddress net.IP
+		localPort    uint16
+		err          error
+	)
+
+	for scanner.Scan() {
+		fields := strings.Fields(string(scanner.Text()))
+		switch {
+		case fields[0] == "IP" || fields[0] == "Prot" || fields[1] == "RemoteAddress:Port":
+			continue
+		case fields[0] == "TCP" || fields[0] == "UDP":
+			proto = fields[0]
+			localAddress, localPort, err = parseIpPort(fields[1])
+			if err != nil {
+				return nil, err
+			}
+		case fields[0] == "->":
+			remoteAddress, remotePort, err := parseIpPort(fields[1])
+			if err != nil {
+				return nil, err
+			}
+			weight, err := strconv.ParseUint(fields[3], 10, 16)
+			if err != nil {
+				return nil, err
+			}
+			activeConn, err := strconv.ParseUint(fields[4], 10, 16)
+			if err != nil {
+				return nil, err
+			}
+			inActConn, err := strconv.ParseUint(fields[5], 10, 16)
+			if err != nil {
+				return nil, err
+			}
+			status = append(status, ipvsBackendStatus{
+				localAddress:  localAddress,
+				localPort:     localPort,
+				remoteAddress: remoteAddress,
+				remotePort:    remotePort,
+				proto:         proto,
+				weight:        uint16(weight),
+				activeConn:    uint16(activeConn),
+				inActConn:     uint16(inActConn),
+			},
+			)
+		}
+	}
+	return status, nil
+}
+
+func getIpvsStats() (map[string]uint64, error) {
+	file, err := os.Open(procNetIpvsStats)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	return parseIpvsStats(file)
+}
+
+func parseIpvsStats(file io.Reader) (map[string]uint64, error) {
+	var (
+		statContent []byte
+		statFields  []string
+		stats       = map[string]uint64{}
+	)
+
+	statContent, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, err
+	}
+
+	statFields = strings.Fields(strings.SplitN(string(statContent), "\n", 4)[2])
+
+	for i := range ipvsStatFieldNames {
+		stats[ipvsStatFieldNames[i]], err = strconv.ParseUint(statFields[i], 16, 64)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return stats, nil
+}
+
+func parseIpPort(s string) (net.IP, uint16, error) {
+	tmp := strings.SplitN(s, ":", 2)
```

Comment:
Could you check here if `tmp` has the expected length and return an error if not? Otherwise the line below will crash the node exporter in case that ever isn't true.